Dfe.Academies.External.Web/Routing/HyphenateRouteParameterTransformer.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Text.RegularExpressions;
+
+namespace Dfe.Academies.External.Web.Routing
+{
+    public class HyphenateRouteParameterTransformer : IOutboundParameterTransformer
+    {
+        public string? TransformOutbound(object? value)
+        {
```

Comment:
would that work

```
var stringValue = value as string;

return stringValue == null ? null : Regex.Replace(stringValue, "([a-z])([A-Z])", "$1-$2").ToLower(); 
```

Reply:
I played around with at a bit and Visual Studio gave me a tip "use pattern matching" so I've done that instead.